r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/MySqlTypeMetadata.java

```diff
@@ -117,20 +133,22 @@ public boolean equals(Object o) {
 
         MySqlTypeMetadata that = (MySqlTypeMetadata) o;
 
-        return typeId == that.typeId && definitions == that.definitions && collationId == that.collationId;
+        return typeId == that.typeId && definitions == that.definitions && collationId == that.collationId && 
+        		Objects.equals(extendedTypeInfo, that.extendedTypeInfo);
     }
 
     @Override
     public int hashCode() {
         int result = 31 * typeId + (int) definitions;
-        return 31 * result + collationId;
+        return 31 * result + collationId + extendedTypeInfo.hashCode();
```

Comment:
Looks like it's not null-safe. Could you make it null-safe and apply *31 as well?
ex)
```
        result =  31 * result + collationId;
        return 31 * result + (extendedTypeInfo == null ? 0 : extendedTypeInfo.hashcode)
```